core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/evaluator/PromptEvaluator.kt

```diff
@@ -0,0 +1,161 @@
+package com.xebia.functional.xef.prompt.evaluator
+
+import com.xebia.functional.xef.auto.CoreAIScope
+import com.xebia.functional.xef.auto.PromptConfiguration
+import com.xebia.functional.xef.llm.Chat
+import com.xebia.functional.xef.llm.models.chat.Message
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+object PromptEvaluator {
+
+  @Serializable
+  data class Score(
+    val completeness: Double,
+    val accuracy: Double,
+    val efficiency: Double,
+    val robustness: Double,
+    val biasAndFairness: Double,
+    val interpretability: Double,
+    val usability: Double,
+    val safetyAndSecurity: Double,
+    val creativityAndFlexibility: Double,
+    val internationalization: Double,
+    val scalability: Double,
+    val legalAndEthicalConsiderations: Double,
+    val reasoning: String,
```

Comment:
I think this should be a list of `ScoreCriteria` instead, but happy to make that change in the following PR.

Reply:
I tried, but I changed the serialization model into a regular chat model to support gpt3.5 16k, which would allow larger prompts. This is why the object returned is so simple, but we may be able to make it work with a few tweaks to the prompt functions